app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  has_many :posts
+
+  email_validor = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :username, :email, :password, presence: true
+  validates :username, :email, uniqueness: true
+  validates :username, length: { in: 4..12 }
+  validates :password, length: { in: 6..16 }
+  validates :email,
+            format: {
+              with: email_validor,
+              message: 'this is not a valid email'
+            }
```

Comment:
- **[SUGGESTION]** Validating each column separately in one line is much better and easier to read, but I understand if you are facing issues with rubocop.

Reply:
**Thank you for Ahmed your review!** 
It was helpful, we also added the **dependent: :destroy** in the **has_many** method to be able to delete post and comments related to one user.